src/rez/package_cache.py

```diff
@@ -214,17 +230,43 @@ def add_variant(self, variant, force=False):
                     % package.repository
                 )
 
-        no_op_statuses = (
+        no_op_statuses = {
             self.VARIANT_FOUND,
-            self.VARIANT_COPYING,
-            self.VARIANT_COPY_STALLED
-        )
+            self.VARIANT_COPY_STALLED,
+        }
+        if not wait_for_copying:
+            # Copying variants are only no-ops if we want to ignore them.
+            no_op_statuses.add(self.VARIANT_COPYING)
 
         # variant already exists, or is being copied to cache by another thread/proc
         status, rootpath = self._get_cached_root(variant)
         if status in no_op_statuses:
+            if logger:
+                logger.warning(f"Not caching {variant.qualified_name}. "
+                               f"Variant {self.STATUS_DESCRIPTIONS[status]}")
             return (rootpath, status)
 
+        if wait_for_copying and status == self.VARIANT_COPYING:
+            spinner = PixelSpinner(f"Waiting for {variant.qualified_name} to finish copying. ")
+            while status == self.VARIANT_COPYING:
+                spinner.next()
+                time.sleep(self._COPYING_TIME_INC)
+                status, rootpath = self._get_cached_root(variant)
+            else:
+                # Status has changed, so report the change and return
+                if logger:
+                    if status in no_op_statuses:
+                        logger.warning(f"{variant.qualified_name} "
+                                       f"{self.STATUS_DESCRIPTIONS[status]}")
+                    elif status == self.VARIANT_FOUND:
+                        # We have resolved into a satisfactory state
+                        logger.info(f"{variant.qualified_name} "
+                                    f"{self.STATUS_DESCRIPTIONS[status]}")
+                    else:
+                        logger.warning(f"{variant.qualified_name} "
+                                       f"{self.STATUS_DESCRIPTIONS[status]}")
```

Comment:
```suggestion
                    if status == self.VARIANT_FOUND:
                        # We have resolved into a satisfactory state
                        logger.info(f"{variant.qualified_name} "
                                    f"{self.STATUS_DESCRIPTIONS[status]}")
                    else:
                        logger.warning(f"{variant.qualified_name} "
                                       f"{self.STATUS_DESCRIPTIONS[status]}")
```
If I'm not mistaken, this would have the same result.